src/gui/plugins/plotting/Transport.hh

```diff
@@ -0,0 +1,233 @@
+#ifndef TRANSPORT_HH
+#define TRANSPORT_HH
+#include <QObject>
+#include <QString>
+#include <bits/stdc++.h>
+#include <ignition/transport.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/MessageInfo.hh>
+#include <ignition/transport/Publisher.hh>
+
+#include <QVariant>
+#include <QStandardItem>
+#include <QStandardItemModel>
+
+#include "TopicsModel.hh"
+
+#define NONE "none"
+#define FLOAT "float"
+#define INT "int"
+#define DOUBLE "double"
+using namespace std;
+
+class TransportPrivate
+{
+public:
+    /// \brief Node for Commincation
+    ignition::transport::Node node;
+    /// \brief topic to subscribe
+    std::string topic;
+    /// \brief Model to create it from the available topics and messages
+    TopicsModel* model;
+
+    /// \brief integer value if we subscribe to a topic with int msg
+    int intValue;
+    /// \brief float value if we subscribe to a topic with float msg
+    float floatValue;
+    /// \brief double value if we subscribe to a topic with double msg
+    double doubleValue;
+    /// \brief type of subscribed msg (int or float or double)
+    std::string type ;
+    /// \brief supported types to subscribe (int , float ,double)
+    vector<string> supportedTypes;
+};
+
+/// \brief handle Transport Topics Subscribing for one object(Chart) (will extend)
+class Transport
+{
+private:
+    unique_ptr<TransportPrivate> data_ptr;
+
+public:
+    /// \brief Constructor
+    Transport() : data_ptr(new TransportPrivate)
+    {
+        cout << "transport" << endl;
+        this->data_ptr->type = NONE;
+        this->data_ptr->supportedTypes.push_back("ignition.msgs.Float");
+        this->data_ptr->supportedTypes.push_back("ignition.msgs.Double");
+        this->data_ptr->supportedTypes.push_back("ignition.msgs.Int32");
+
+        this->data_ptr->model = new TopicsModel();
+        this->InitModel();
+    }
+
+    /// \brief Destructor
+    ~Transport()
+    {
+        cout << "destrcutor transport" << endl;
```

Comment:
Let's remove these print statements if they're not needed anymore.

For messages that are still wanted, use `ignerr` / `ignwarn` / `ignmsg` / `igndbg`, so that users can opt to see them according to the verbosity level. For example, `ign gazebo -v 4` shows all of them.